www/polj10/sp2/js/pastebin-rest.js

```diff
@@ -0,0 +1,531 @@
+    const password = document.getElementById("loginPassword").value;
+
+    createApiUserKey(userName, password);
+    closeStatusPanel();
+}
+function logout() {
+    const localStorage = window.localStorage;
+    localStorage.removeItem("api_user_key");
+    showAnonymousUserId();
+}
+
+function createPasteElement(paste_key, paste_date, paste_title, paste_expire_date, paste_private, paste_format_long, paste_url) {
+    const pasteElement = document.createElement("li");
+    pasteElement.classList.add("pasteListElement");
+
+    const linkElement = document.createElement("a");
+    const dateElement = document.createElement("span");
+    const titleElement = document.createElement("span");
+    const expireElement = document.createElement("span");
+    const privateElement = document.createElement("em");
+    const formatElement = document.createElement("format");
+    const clearElement = document.createElement("div");
+    const clearElement2 = document.createElement("div");
+    const deleteButton = document.createElement("button");
+
+    linkElement.innerText = paste_url;
+    linkElement.href = paste_url;
+    linkElement.target = "_blank";
+
+
+    titleElement.classList.add("pasteListName");
+    expireElement.classList.add("pasteListDate");
+    dateElement.classList.add("pasteListDate");
+    clearElement.classList.add("clear");
+    clearElement2.classList.add("clear");
+    deleteButton.classList.add("deletePaste");
+    deleteButton.innerText = "Odstranit";
+    deleteButton.onclick = function () {
+        deletePaste(paste_key);
+        listUserPastes();
+    };
+
+    dateElement.innerText = "Vytvořeno" +new Date(paste_date * 1000).toLocaleString("cs-CZ");
+
+
+    titleElement.innerText = paste_title;
+    expireElement.innerText = "Expirace: " + new Date(paste_expire_date * 1000).toLocaleString("cs-CZ");
+    privateElement.innerText = " (" + globals.visibility_paste_code_names[paste_private] + ")";
+    formatElement.innerText = " - " + paste_format_long;
+
+    pasteElement.appendChild(titleElement);
+    pasteElement.appendChild(formatElement);
+    pasteElement.appendChild(privateElement);
+
+
+    pasteElement.appendChild(dateElement);
+    pasteElement.appendChild(clearElement);
+
+    pasteElement.appendChild(linkElement);
+
+    pasteElement.appendChild(expireElement);
+    pasteElement.appendChild(clearElement2);
+    pasteElement.appendChild(deleteButton);
+
+    return pasteElement;
+}
+
+//API FUNCTIONS
+function listUserPastes() {
+    const apiUserKey = localStorage.getItem("api_user_key");
+    const data = new URLSearchParams();
+    data.append("api_dev_key", globals.api_dev_key);
+    data.append("api_user_key", apiUserKey);
+    data.append("api_option", globals.api_option_list);
+    document.getElementById("userPastes").classList.remove("hidden");
+
+    postData(globals.create_paste_url, data,
+        function (responseText) {
+            if (responseText === "No pastes found."){
```

Comment:
neni tam nejaky response code? podle response textu to neni bezpecne

Reply:
Teď z hlavy si nejsem jistý, ale myslím, že je i tomto případě response code 200. A nejen v tomhle případě. Některé odpovědi bohužel ani nemají v dokumentaci a narazil jsem na ně náhodou. Konkrétně situace, kdy překročím počet requestů za určité časové období.